manim/utils/tex_file_writing.py

```diff
@@ -42,67 +45,88 @@ def generate_tex_file(expression, tex_template, source_type):
+            compiler["command"],
+            "-interaction=batchmode",
+            '-output-format="' + compiler["output_format"][1:] + '"',
+            "-halt-on-error",
+            '-output-directory="{}"'.format(tex_dir),
+            '"{}"'.format(tex_file),
+            ">",
+            os.devnull,
+        ]
+    elif compiler["command"] == "xelatex":
+        if compiler["output_format"] == ".xdv":
+            outflag = "-no-pdf"
+        elif compiler["output_format"] == ".pdf":
+            outflag = ""
+        else:
+            raise ValueError("xelatex output is either pdf or xdv")
+        commands = [
+            "xelatex",
+            outflag,
+            "-interaction=batchmode",
+            "-halt-on-error",
+            '-output-directory="{}"'.format(tex_dir),
+            '"{}"'.format(tex_file),
+            ">",
+            os.devnull,
+        ]
+    else:
+        raise ValueError(f"Tex compiler {compiler['command']} unknown.")
+    return " ".join(commands)
+
+
+def tex_to_dvi(tex_file, use_ctex=False, tex_compiler=None):
+    if tex_compiler is None:
+        tex_compiler = (
+            {"command": "xelatex", "output_format": ".xdv"}
+            if use_ctex
+            else {"command": "latex", "output_format": ".dvi"}
+        )
+    result = tex_file.replace(".tex", tex_compiler["output_format"])
     result = Path(result).as_posix()
     tex_file = Path(tex_file).as_posix()
     tex_dir = Path(file_writer_config["tex_dir"]).as_posix()
     if not os.path.exists(result):
-        commands = (
-            [
-                "latex",
-                "-interaction=batchmode",
-                "-halt-on-error",
-                '-output-directory="{}"'.format(tex_dir),
-                '"{}"'.format(tex_file),
-                ">",
-                os.devnull,
-            ]
-            if not use_ctex
-            else [
-                "xelatex",
-                "-no-pdf",
-                "-interaction=batchmode",
-                "-halt-on-error",
-                '-output-directory="{}"'.format(tex_dir),
-                '"{}"'.format(tex_file),
-                ">",
-                os.devnull,
-            ]
-        )
-        exit_code = os.system(" ".join(commands))
+        command = tex_compilation_command(tex_compiler, tex_file, tex_dir)
+        exit_code = os.system(command)
         if exit_code != 0:
             log_file = tex_file.replace(".tex", ".log")
-            raise Exception(
-                (
-                    "LaTeX error converting to dvi. "
-                    if not use_ctex
-                    else "XeLaTeX error converting to xdv. "
-                )
-                + f"See log output above or the log file: {log_file}"
+            raise ValueError(
+                f"{tex_compiler['command']} error converting to"
+                f" {tex_compiler['output_format'][1:]}. See log output above or"
+                f" the log file: {log_file}"
```

Comment:
Why should it be `ValueError`? I think it should be some other thing.

Reply:
I had Exception, @behackl (iirc) suggested a ValueError in a previous review.
I have no preference here.

Tex Errors are Tex Errors.... and we don't handle them gracefully in any case.

Reply:
While don't create a small class and raise them?  https://stackoverflow.com/questions/1319615/proper-way-to-declare-custom-exceptions-in-modern-python#1319675

Reply:
sigh. why? 
- Well I've never done it before. 
- I had no way to know to do it. 
- I was reusing upstream code: https://github.com/ManimCommunity/manim/blob/master/manim/utils/tex_file_writing.py#L76
- I'm tired.
... but it's probably a good suggestion in general.
Moreover I think a lot can be said for cleaning up our tex-compilation-error-handling overall. We don't even distinguish between a tex-syntax error and a file-not-found error. Perhaps this can be another issue?

Reply:
I think it's fine as a ValueError for now.

Reply:
Yes, leave if for now and make a new issue.